serialization.go

```diff
@@ -248,60 +248,95 @@ func IsPreIssuer(issuer *x509.Certificate) bool {
 	return false
 }
 
-// LogEntryFromLeaf converts a LeafEntry object (which has the raw leaf data after JSON parsing)
-// into a LogEntry object (which includes x509.Certificate objects, after TLS and ASN.1 parsing).
-// Note that this function may return a valid LogEntry object and a non-nil error value, when
-// the error indicates a non-fatal parsing error.
-func LogEntryFromLeaf(index int64, leafEntry *LeafEntry) (*LogEntry, error) {
-	var leaf MerkleTreeLeaf
-	if rest, err := tls.Unmarshal(leafEntry.LeafInput, &leaf); err != nil {
-		return nil, fmt.Errorf("failed to unmarshal MerkleTreeLeaf for index %d: %v", index, err)
+// RawLogEntryFromLeaf TLS-unmarshals the passed in LeafEntry to a RawLogEntry.
```

Comment:
nit: The comment is too implementation based; a user hopefully shouldn't have to care about what encoding methods are in use inside.

(Compare with the `LogEntryFromLeaf` comment above: dropping the parenthetical remarks (which do go into implementation details), it "..converts a LeafEntry object into a LogEntry object" -- which tells me what I need to know as an API user.)

Reply:
Done (but still left some impl details in the comment).